components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/VariantListController.java

```diff
@@ -0,0 +1,274 @@
+    {
+        return CONTROLLER_NAME;
+    }
+
+    @Override
+    protected List<String> getProperties()
+    {
+        return Arrays.asList(VARIANT_KEY, GENESYMBOL_KEY, PROTEIN_KEY, TRANSCRIPT_KEY, DBSNP_KEY, ZYGOSITY_KEY,
+            EFFECT_KEY,
+            INTERPRETATION_KEY, INHERITANCE_KEY, EVIDENCE_KEY, SEGREGATION_KEY, SANGER_KEY, RESOLUTION_KEY);
+    }
+
+    @Override
+    protected List<String> getBooleanFields()
+    {
+        return Collections.emptyList();
+    }
+
+    @Override
+    protected List<String> getCodeFields()
+    {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public PatientData<Map<String, String>> load(Patient patient)
+    {
+        try {
+            XWikiDocument doc = (XWikiDocument) this.documentAccessBridge.getDocument(patient.getDocument());
+            List<BaseObject> variantXWikiObjects = doc.getXObjects(VARIANT_CLASS_REFERENCE);
+            if (variantXWikiObjects == null || variantXWikiObjects.isEmpty()) {
+                return null;
+            }
+
+            List<Map<String, String>> allVariants = new LinkedList<Map<String, String>>();
+            for (BaseObject variantObject : variantXWikiObjects) {
+                if (variantObject == null || variantObject.getFieldList().size() == 0) {
+                    continue;
+                }
+                Map<String, String> singleVariant = new LinkedHashMap<String, String>();
+                for (String property : getProperties()) {
+                    String value = getFieldValue(variantObject, property);
+                    if (value == null) {
+                        continue;
+                    }
+                    singleVariant.put(property, value);
+                }
+                allVariants.add(singleVariant);
+            }
+            return new IndexedPatientData<Map<String, String>>(getName(), allVariants);
+        } catch (Exception e) {
+            this.logger.error("Could not find requested document or some unforeseen "
+                + "error has occurred during controller loading ", e.getMessage());
+        }
+        return null;
+    }
+
+    private String getFieldValue(BaseObject variantObject, String property)
+    {
+        if (EVIDENCE_KEY.equals(property)) {
+            StringListProperty fields = (StringListProperty) variantObject.getField(property);
+            if (fields == null || fields.getList().size() == 0) {
+                return null;
+            }
+            // return ListClass.getStringFromList(fields.getList(), ListClass.DEFAULT_SEPARATOR);
+            return fields.getTextValue();
+
+        } else {
+            BaseStringProperty field = (BaseStringProperty) variantObject.getField(property);
+            if (field == null) {
+                return null;
+            }
+            return field.getValue();
+        }
+    }
+
+    private void removeKeys(Map<String, String> item, List<String> keys, Map<String, String> enablingProperties,
```

Comment:
Is this function not identical to the one in the file above? Maybe factor it out to a separate file?